src/lib.rs

```diff
@@ -569,19 +602,39 @@ impl Config {
         for file in self.files.iter() {
             let obj = dst.join(file).with_extension("o");
             let obj = if !obj.starts_with(&dst) {
-                dst.join(obj.file_name().unwrap())
+                dst.join(match obj.file_name() {
+                    Some(s) => s,
+                    None => return Err(Error {
+                        kind: ErrorKind::IOError,
+                        message: "Getting object file details failed.".to_string(),
+                    }),
+                })
```

Comment:
This might be a little neater as a helper function, so you could do something like:

```rust
dst.join(obj.file_name().ok_or_else(|| err("failed to get details"))?);
```

(or something like that)